### PR TITLE
feat: update in logging to console and Unify case usage, seperators and punctuation for logging

### PR DESCRIPTION
**Pull request by mihirm21**
Signed-off-by: Mihir Mittal <105881639+mihirm21@users.noreply.github.com>

### Related Issues

[FEATURE] Pretty-print logging #237 
[REFACTOR] Unify case usage, seperators and punctuation for logging #298

### CODE REVIEW

**Comment by mihirm21** [comment]
Please check this commit.

File: cmd/start.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ const (
 	bearerTokenFlagName    = "bearer-token"
 	corsFlagName           = "cors-origin"
 	syncProviderFlagName   = "sync-provider"
+	prettyLogFlagName      = "prettylogger"
```
Comment (james-milligan) [inline cmd/start.go:27]: can we update this flag name to something a little clearer, such as `log-format`, with this value a developer would have to pass `prettyLogger=json`

File: cmd/start.go [hunk 2]
```diff
@@ -55,6 +56,8 @@ func init() {
 	flags.StringP(
 		syncProviderFlagName, "y", "", "DEPRECATED: Set a sync provider e.g. filepath or remote",
 	)
+	flags.StringP(prettyLogFlagName, "z", "console", "Set a logger with console default "+
```
Comment (james-milligan) [inline cmd/start.go:59]: `Set the logging format, e.g. console or json`

**Comment by james-milligan** [comment]
It looks like the PR is failing on linting and tests, please can you resolve these issues.

It also might be more appropriate to move the changes from commit `5081d69ca6f404d3c0a06ebacc006f63c16ef057` into a separate PR as these changes resolve a separate issue.

**Comment by mihirm21** [comment]
plz check now. I have solved two issues in this one #237 and #298.

File: cmd/start.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ const (
 	bearerTokenFlagName    = "bearer-token"
 	corsFlagName           = "cors-origin"
 	syncProviderFlagName   = "sync-provider"
+	prettyLogFlagName      = "log-format"
```
Comment (beeme1mr) [inline cmd/start.go:27]:
```suggestion
	logFormatFlagName      = "log-format"
```

Please also update all the references to this variable.

File: cmd/start.go [hunk 4]
```diff
@@ -83,14 +86,22 @@ var startCmd = &cobra.Command{
...
-		l, err := logger.NewZapLogger(level)
+		l, err := logger.NewZapLogger(level, viper.GetString(prettyLogFlagName))
 		if err != nil {
 			log.Fatalf("can't initialize zap logger: %v", err)
 		}
 		logger := logger.NewLogger(l, Debug)
 		rtLogger := logger.WithFields(zap.String("component", "start"))
 
 		if viper.GetString(syncProviderFlagName) != "" {
+			rtLogger.Warn("DEPRECATED: the --sync-provider flag has been deprecated " +
```
Comment (beeme1mr) [inline cmd/start.go:97]: It looks like this was duplicated during a rebase. Could you please doublecheck this section?

**Review by beeme1mr — CHANGES_REQUESTED** [review]

**Comment by beeme1mr** [comment]
Also, please run `make generate-docs` to update the documentation.

File: cmd/start.go (hunk 3 above)
Comment (mihirm21) [reply cmd/start.go]: done

File: cmd/start.go (hunk 4 above)
Comment (mihirm21) [reply cmd/start.go:97]: done
btw, could you tell me why it could occur every time I committed?

**Comment by mihirm21** [comment]
plz review it

File: cmd/start.go [hunk 5]
```diff
@@ -83,23 +86,22 @@ var startCmd = &cobra.Command{
...
-			rtLogger.Warn("DEPRECATED: The --sync-provider flag has been deprecated. " +
-				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md")
+			rtLogger.Warn("DEPRECATED: the --sync-provider flag has been deprecated " +
+				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration.md")
 		}
 
-		if viper.GetString(evaluatorFlagName) != "json" {
-			rtLogger.Warn("DEPRECATED: The --evaluator flag has been deprecated. " +
-				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md")
+		if viper.GetString(evaluatorFlagName) != "" {
```
Comment (james-milligan) [inline cmd/start.go:101]: should be `viper.GetString(evaluatorFlagName) != "json"`

File: pkg/logger/logger.go [hunk 6]
```diff
@@ -188,6 +188,9 @@ func NewZapLogger(level zapcore.Level) (*zap.Logger, error) {
 		},
 		DisableCaller: false,
 	}
+	if cfg.Encoding == "" {
```
Comment (james-milligan) [inline pkg/logger/logger.go:191]: these lines can be removed as the default is already set via the flag definition itself:
```
flags.StringP(logFormatFlagName, "z", "console", "Set the logging format, e.g. console or json ")
```

File: cmd/start.go [hunk 7]
```diff
@@ -83,23 +86,22 @@ var startCmd = &cobra.Command{
...
 			log.Fatalf("can't initialize zap logger: %v", err)
 		}
 		logger := logger.NewLogger(l, Debug)
 		rtLogger := logger.WithFields(zap.String("component", "start"))
 
 		if viper.GetString(syncProviderFlagName) != "" {
-			rtLogger.Warn("DEPRECATED: The --sync-provider flag has been deprecated. " +
-				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md")
+			rtLogger.Warn("DEPRECATED: the --sync-provider flag has been deprecated " +
+				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration.md")
```
Comment (james-milligan) [inline cmd/start.go:98]: This link as been broken, should be `https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md`

File: cmd/start.go [hunk 8]
```diff
@@ -83,23 +86,22 @@ var startCmd = &cobra.Command{
...
 		}
 
-		if viper.GetString(evaluatorFlagName) != "json" {
-			rtLogger.Warn("DEPRECATED: The --evaluator flag has been deprecated. " +
-				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md")
+		if viper.GetString(evaluatorFlagName) != "" {
+			rtLogger.Warn("DEPRECATED: the --evaluator flag has been deprecated " +
+				"Docs: https://github.com/open-feature/flagd/blob/main/docs/configuration.md")
```
Comment (james-milligan) [inline cmd/start.go:103]: same for this one: 
`https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md`

File: pkg/sync/file/filepath_sync.go [hunk 9]
```diff
@@ -46,20 +46,18 @@ func (fs *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
...
-				fs.Logger.Info("Filepath notifier closed")
+				fs.Logger.Info("filepath notifier closed")
 				return errors.New("filepath notifier closed")
 			}
 
-			fs.Logger.Info(fmt.Sprintf("Filepath event: %s %s", event.Name, event.Op.String()))
+			fs.Logger.Info(fmt.Sprintf("filepath event: %s %s", event.Name, event.Op.String()))
 
 			switch event.Op {
-			case fsnotify.Create, fsnotify.Write:
```
Comment (james-milligan) [inline pkg/sync/file/filepath_sync.go:61]: please revert this change

**Comment by mihirm21** [comment]
@james-milligan plz see now

File: pkg/sync/file/filepath_sync.go (hunk 9 above)
Comment (james-milligan) [reply pkg/sync/file/filepath_sync.go:61]: https://github.com/open-feature/flagd/actions/runs/4032086735/jobs/6949951527

The code had been edited, likely during a merge but is resolved now

**Review by james-milligan — APPROVED** [review]
Nice work, thank you for your contribution!

**Review by beeme1mr — APPROVED** [review]

**Comment by beeme1mr** [comment]
The OFO should be updated once this feature is available.

https://github.com/open-feature/open-feature-operator/issues/334

**Comment by beeme1mr** [comment]
Thank you for your contribution @mihirm21! This feature will be available in the [next release](https://github.com/open-feature/flagd/pull/343).
